distributed/protocol/h5py.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import print_function, division, absolute_import
+
+import h5py
+
+from .serialize import register_serialization
+
+from ..utils import log_errors, ensure_bytes
+
+
+def serialize_h5py_file(f):
+    return {'filename': f.filename, 'mode': f.mode}, []
```

Comment:
That would be my inclination, yes. Require `mode in {'r', 'r+'}`.

But what then do you do with writeable files to ensure they don't get passed around?

h5py does have support for "parallel HDF5" using MPI but it seems a good deal more complex (I've never used it):
http://docs.h5py.org/en/latest/mpi.html